### PR TITLE
Migrated `pkg/scheduler/framework/plugins/noderesources` to contextual logging

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Migrated `pkg/scheduler/framework/plugins/noderesources` to contextual logging

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/91633#issuecomment-1467645065

#### Does this PR introduce a user-facing change?

```release-note
Migrated the noderesources scheduler plugin to use [contextual logging](https://k8s.io/docs/concepts/cluster-administration/system-logs/#contextual-logging).
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging
```
/sig instrumentation
/wg structured-logging
/area logging
/sig scheduling

### CODE REVIEW

File: pkg/scheduler/framework/plugins/noderesources/resource_allocation.go [hunk 1]
```diff
@@ -70,8 +71,8 @@ func (r *resourceAllocationScorer) score(
 
 	score := r.scorer(requested, allocatable)
 
-	if klogV := klog.V(10); klogV.Enabled() { // Serializing these maps is costly.
-		klogV.InfoS("Listing internal info for allocatable resources, requested resources and score", "pod",
+	if klogV := logger.V(10); klogV.Enabled() { // Serializing these maps is costly.
+		logger.V(10).Info("Listing internal info for allocatable resources, requested resources and score", "pod",
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/noderesources/resource_allocation.go:75]:
```suggestion
		klogV.Info("Listing internal info for allocatable resources, requested resources and score", "pod",
```

Comment (mengjiao-liu) [reply pkg/scheduler/framework/plugins/noderesources/resource_allocation.go]: Done.

File: pkg/scheduler/framework/plugins/noderesources/resource_allocation.go [hunk 2]
```diff
@@ -70,8 +71,8 @@ func (r *resourceAllocationScorer) score(
 
 	score := r.scorer(requested, allocatable)
 
-	if klogV := klog.V(10); klogV.Enabled() { // Serializing these maps is costly.
-		klogV.InfoS("Listing internal info for allocatable resources, requested resources and score", "pod",
+	if loggerV := logger.V(10); loggerV.Enabled() { // Serializing these maps is costly.
+		loggerV.Info("Listing internal info for allocatable resources, requested resources and score", "pod",
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/noderesources/resource_allocation.go:75]:
```suggestion
		loggerV.Info("Listed internal info for allocatable resources, requested resources and score", "pod",
```

File: pkg/scheduler/framework/plugins/noderesources/fit.go [hunk 3]
```diff
@@ -381,5 +382,5 @@ func (f *Fit) Score(ctx context.Context, state *framework.CycleState, pod *v1.Po
 		}
 	}
 
-	return f.score(pod, nodeInfo, s.podRequests)
+	return f.score(klog.FromContext(ctx), pod, nodeInfo, s.podRequests)
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/noderesources/fit.go:385]:
```suggestion
	return f.score(ctx, pod, nodeInfo, s.podRequests)
```
Prefer context than logger as parameters.

File: pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go [hunk 4]
```diff
@@ -105,7 +106,7 @@ func (ba *BalancedAllocation) Score(ctx context.Context, state *framework.CycleS
 	// Detail: score = (1 - std) * MaxNodeScore, where std is calculated by the root square of Σ((fraction(i)-mean)^2)/len(resources)
 	// The algorithm is partly inspired by:
 	// "Wei Huang et al. An Energy Efficient Virtual Machine Placement Algorithm with Balanced Resource Utilization"
-	return ba.score(pod, nodeInfo, s.podRequests)
+	return ba.score(klog.FromContext(ctx), pod, nodeInfo, s.podRequests)
```
Comment (kerthcet) [inline pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go:109]:
```suggestion
	return ba.score(ctx, pod, nodeInfo, s.podRequests)
```

File: pkg/scheduler/framework/plugins/noderesources/resource_allocation.go (hunk 2 above)
Comment (mengjiao-liu) [reply pkg/scheduler/framework/plugins/noderesources/resource_allocation.go]: Done.

File: pkg/scheduler/framework/plugins/noderesources/fit.go (hunk 3 above)
Comment (mengjiao-liu) [reply pkg/scheduler/framework/plugins/noderesources/fit.go]: Pass context as a parameter is also ok. Updated.

File: pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go (hunk 4 above)
Comment (mengjiao-liu) [reply pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go]: Done.

**Review by kerthcet — COMMENTED** [review]
/lgtm
Thanks @mengjiao-liu

**Review by alculquicondor — COMMENTED** [review]
/approve

**Pull request merged**